From: Josh Boyer <jwboyer@fedoraproject.org>
Date: Wed, 2 Apr 2014 11:25:47 -0400
Subject: [PATCH 02/12] net: qlcnic: include irq.h for irq definitions
Patch-mainline: v3.15-rc1
Git-commit: acdd32be6d3ec5f872d9fd930cac54d18d6c74ac
References: bnc#853685 FATE#315935

The qlcnic driver fails to build on ARM with errors like:

In file included from drivers/net/ethernet/qlogic/qlcnic/qlcnic.h:36:0,
                 from drivers/net/ethernet/qlogic/qlcnic/qlcnic_hw.c:8:
drivers/net/ethernet/qlogic/qlcnic/qlcnic_83xx_hw.h:585:1: error: unknown type name 'irqreturn_t'
 irqreturn_t qlcnic_83xx_clear_legacy_intr(struct qlcnic_adapter *);
 ^

Nothing in the driver is explicitly including the irq definitions, so we
add an include of linux/irq.h to pick them up.

Signed-off-by: Josh Boyer <jwboyer@fedoraproject.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/qlogic/qlcnic/qlcnic.h |    1 +
 1 file changed, 1 insertion(+)
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic.h
@@ -23,6 +23,7 @@
 #include <linux/ethtool.h>
 #include <linux/mii.h>
 #include <linux/timer.h>
+#include <linux/irq.h>
 
 #include <linux/vmalloc.h>
 
